src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,50 @@
 package core.basesyntax.service;
 
 import core.basesyntax.dao.StorageDao;
-import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_LOGIN_OR_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE_TO_REGISTER = 18;
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        return userValidation(user) ? storageDao.add(user) : null;
```

Comment:
The `userValidation` method should not return a boolean since the method throws an exception for invalid data. Instead, you can call `userValidation` without a return value and proceed with adding the user to storage if no exception is thrown.